.github/workflows/docs/publish-docs.yml

```
@@ -0,0 +1,107 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Ensure `docs/` exists
+        run: |
+          if [ ! -d "docs" ]; then
+            echo "::error::The 'docs/' folder is missing. This workflow requires documentation to be present."
+            exit 1
+          fi
+      - name: Set Up Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.4)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)